force-app/main/default/classes/DonationHistoryController_TEST.cls

```diff
@@ -0,0 +1,231 @@
+        list<Contact> listConSrc = UTIL_UnitTestData_TEST.listConT; 
+
+        Opportunity opportunityHardCredit = new Opportunity(
+            Name='HardCredit1', 
+            AccountId=accSrc.Id, 
+            Primary_Contact__c=listConSrc[0].Id, 
+            Amount=400, 
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        Opportunity opportunitySoftCredit = new Opportunity(
+            Name='SoftCredit1', 
+            AccountId=accSrc.Id, 
+            Primary_Contact__c=listConSrc[1].Id, 
+            Amount=200, 
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        Opportunity opportunityPartialSoftCredit = new Opportunity(
+            Name='PartialSoftCredit1', 
+            AccountId=accSrc.Id, 
+            Primary_Contact__c=listConSrc[1].Id, 
+            Amount=150, 
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        Opportunity opportunityInfluencer = new Opportunity(
+            Name='InfluencerOpportunity', 
+            AccountId=accSrc.Id, 
+            Primary_Contact__c=listConSrc[1].Id, 
+            Amount=150,
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        List<Opportunity> recordsOpp = new List<Opportunity>();
+        recordsOpp.add(opportunityHardCredit);
+        recordsOpp.add(opportunitySoftCredit);
+        recordsOpp.add(opportunityPartialSoftCredit);
+        recordsOpp.add(opportunityInfluencer);
+        insert recordsOpp;
+
+        OpportunityContactRole oppCntRoleSoftCredit = new OpportunityContactRole(
+            Role='Soft Credit', 
+            OpportunityId=opportunitySoftCredit.Id, 
+            ContactId=listConSrc[0].Id, 
+            isPrimary=false
+        );
+        OpportunityContactRole oppCntRolePartialSoftCredit = new OpportunityContactRole(
+            Role='Soft Credit', 
+            OpportunityId=opportunityPartialSoftCredit.Id, 
+            ContactId=listConSrc[0].Id, 
+            isPrimary=false
+        );
+        OpportunityContactRole oppCntRoleInfluencerCredit = new OpportunityContactRole(
+            Role='Influencer', 
+            OpportunityId=opportunityInfluencer.Id, 
+            ContactId=listConSrc[0].Id, 
+            isPrimary=false
+        );
+        List<OpportunityContactRole> recordsOcr = new List<OpportunityContactRole>();
+        recordsOcr.add(oppCntRoleSoftCredit);
+        recordsOcr.add(oppCntRolePartialSoftCredit);
+        recordsOcr.add(oppCntRoleInfluencerCredit);
+        insert recordsOcr;
+
+        Partial_Soft_Credit__c partialSoftCredit = new Partial_Soft_Credit__c (
+            Amount__c=50,
+            Contact__c=listConSrc[0].Id,
+            Opportunity__c=opportunityPartialSoftCredit.Id,
+            Contact_Role_ID__c=oppCntRolePartialSoftCredit.Id
+        );
+        insert partialSoftCredit;
+        generateFilterRules();
```

Comment:
generateFilterRules() won't work here because it is setting a static variable on another class, which gets cleared between each test run. Instead, it'll need to be called inside of each test method in this class.

Reply:
the method was added for each test as required, and yes the results have to be delivered in order but we have a ticket where the implementation of the infinite scroll and the order will be evaluated, since we perform the filtration after obtaining the data it can generate a variation in how the data is displayed